test-project/Assets/EditmodeTests/Collections/ReferenceProviderTests.cs

```diff
@@ -35,25 +29,30 @@ public void Disposed_world_cleans_provider()
         [Test]
         public void Removed_component_disposes_reference()
         {
+            var template = GetTemplate();
             using (var mockWorld = MockWorld.Create(new MockWorld.Options()))
             {
-                mockWorld.Step(world =>
-                {
-                    world.Connection.CreateEntity(entityId, new EntityTemplate());
-
-                    world.Connection.AddComponent(entityId, Position.ComponentId,
-                        new Position.Update {Coords = Coordinates.Zero});
-
-                    world.Connection.AddComponent(entityId, Metadata.ComponentId,
-                        new Metadata.Update {EntityType = "EntityWithReferenceComponent"});
-                }).Step(world =>
-                {
-                    Assert.AreEqual(1, ReferenceProvider<string>.Count);
-                    world.Connection.RemoveComponent(entityId, Metadata.ComponentId);
-                });
+                mockWorld
+                    .Step(world =>
+                    {
+                        world.Connection.CreateEntity(entityId, template);
+                    })
+                    .Step(world =>
+                    {
+                        Assert.AreEqual(1, ReferenceProvider<string>.Count);
+                        world.Connection.RemoveEntityAndComponents(entityId, template);
```

Comment:
The test previously only removed the component, not the whole entity?

Reply:
Discussed offline - this is what created the unhappiness in the first place (the fact that `RemoveEntity` didn't work originally). So we are good to swap this over